pkg/unshare/unshare_linux.go

```diff
@@ -76,6 +77,36 @@ func getRootlessGID() int {
 	return os.Getegid()
 }
 
+type idtype int64
+
+const (
+	uid idtype = 0
+	gid        = 1
+)
+
+func isSetID(id idtype, path string) (bool, error) {
```

Comment:
This would be a bit shorter if the callers passed in the mode bit and the capability to look for instead of an enumerated value.

Reply:
Fixed.